src/equations/hyperbolic_serre_green_naghdi_1d.jl

```diff
@@ -139,9 +139,14 @@ function set_approximation_variables!(q, mesh,
     # H ≈ h
     @.. H = eta + D - equations.eta0 # h = (h + b) + (eta0 - b) - eta0
 
-    # w ≈ -h v_x
+    # w ≈ -h v_x + 3/2 v b_x
     mul!(w, D1, v)
     @.. w = -H * w
+    if !(equations.bathymetry_type isa BathymetryFlat)
+        b = bathymetry(q, equations)
+        b_x = D1 * b
```

Comment:
Unfortunately, we do not have access to `cache` here. So it's difficult to do the multiplication in-place. Do you see a way for an in-place multiplication or would you be fine with this version since it is also not so performance-critical (it's only computed once and not in `rhs!`), @ranocha?